tests/acceptance/features/bootstrap/RoleAssignmentContext.php

```diff
@@ -0,0 +1,246 @@
+		// Get all the contexts you need in this context from here
+		$this->featureContext = $environment->getContext('FeatureContext');
+		$this->spacesContext = $environment->getContext('SpacesContext');
+		$this->baseUrl = \trim($this->featureContext->getBaseUrl(), "/");
+	}
+
+	/**
+	 * @When /^user "([^"]*)" tries to get all existing roles$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function getAllExistingRoles(string $user): void {
+		$fullUrl = $this->baseUrl . $this->setttingsUrl . "roles-list";
+		$this->featureContext->setResponse($this->spacesContext->sendPostRequestToUrl($fullUrl, $user, $this->featureContext->getPasswordForUser($user), "{}"));
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $userId
+	 * @param string $roleId
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function sendRequestToAssignRoleToUser(string $user, string $userId, string $roleId): void {
+		$fullUrl = $this->baseUrl . $this->setttingsUrl . "assignments-add";
+		$body = json_encode(["account_uuid" => $userId, "role_id" => $roleId], JSON_THROW_ON_ERROR);
+
+		$this->featureContext->setResponse($this->spacesContext->sendPostRequestToUrl($fullUrl, $user, $this->featureContext->getPasswordForUser($user), $body));
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $userId
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function sendRequestAssignmentsList(string $user, string $userId): void {
+		$fullUrl = $this->baseUrl . $this->setttingsUrl . "assignments-list";
+		$body = json_encode(["account_uuid" => $userId], JSON_THROW_ON_ERROR);
+
+		$this->featureContext->setResponse($this->spacesContext->sendPostRequestToUrl($fullUrl, $user, $this->featureContext->getPasswordForUser($user), $body));
+	}
+
+	/**
+	 * @When /^the administrator has given "([^"]*)" the role "([^"]*)" using the settings api$/
+	 *
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function theAdministratorHasGivenUserTheRole(string $user, string $role): void {
+		$admin = $this->featureContext->getAdminUserName();
+		$roleId = $this->userGetRoleIdByRoleName($admin, $role);
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
+		$this->setRoleToUser($admin, $userId, $roleId);
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return string
+	 */
+	public function userGetRoleIdByRoleName($user, $role): string {
+		$this->getAllExistingRoles($user);
+
+		if ($this->featureContext->getResponse()) {
+			$rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
+			if (isset(\json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["bundles"])) {
+				$bundles = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["bundles"];
+			}
```

Comment:
I think we should do something if the bundles key is not in the array
```suggestion
                        $decodedBody = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR);
			Assert::assertArrayHasKey(
			'bundles',
			$decodedBody,
			__METHOD__ . " could not find bundles in body")
	   	        $bundles = $decodedBody["bundles"];
```